server/listener.go

```diff
@@ -120,7 +120,13 @@ func (l *Listener) Accept() (net.Conn, error) {
 }
 
 func (l *Listener) Close() error {
+	if l.shutdown == nil {
+		return nil
+	}
+
 	close(l.shutdown)
+	l.shutdown = nil
```

Comment:
can Close() every be called several times? cause then you might have a race, and this could still panic

Reply:
looks like there is two listeners that can concurrently call Close, maybe wrap these in a mutex? or bake into whatever concurrency control Listener also uses?

Reply:
Initially, it was not being called multiple times, but since we want to stop the server when the `.dolt` dir removed during running server, we call this method before serverController calls this Close() method.

Reply:
Good point about the concurrency; wrapping access in a mutex seems prudent.